src/bitcoin-cli.cpp

```diff
@@ -97,11 +97,19 @@ static bool AppInitRPC(int argc, char* argv[])
 
 UniValue CallRPC(const string& strMethod, const UniValue& params)
 {
-    if (mapArgs["-rpcuser"] == "" && mapArgs["-rpcpassword"] == "")
-        throw runtime_error(strprintf(
-            _("You must set rpcpassword=<password> in the configuration file:\n%s\n"
-              "If the file does not exist, create it with owner-readable-only file permissions."),
-                GetConfigFile().string().c_str()));
+    std::string strRPCUserColonPass;
+    if (mapArgs["-rpcpassword"] == "") {
+        /* Try fall back to cookie-based authentication if no password is provided */
+        if (!GetAuthCookie(&strRPCUserColonPass)) {
+            throw runtime_error(strprintf(
```

Comment:
This error should be changed. It will be shown if bitcoin-cli is used while bitcoind is not running, so it's confusing to tell the user to just set a password (which will indeed make this message go away, but just enough to get to `connection refused` :-).